### DISCUSSION TITLE
update_control() for group border control

### DISCUSSION THREAD

**Discussion by LucasMariniFalbo** [elementor/elementor#25695] [Developers Community]
### Start a discussion

So, I have the following code that register a border control. Works just fine 
`$this->add_group_control(
Group_Control_Border::get_type(),
	[
		'name'           => 'item_border',
		'label'          => esc_html__( 'Border', 'text-domain' ),
		'fields_options' => [
			'border' => [
				'default' => 'solid',
			],
			'width'  => [
				'default' => [
					'top'      => '1',
					'right'    => '1',
					'bottom'   => '1',
					'left'     => '1',
					'isLinked' => true,
				],
			],
			'color' => [
				'default' => '#EEE'
			]
		],
		'selector'       => '{{WRAPPER}} .ui-e-item',
	]
);`

However, I also have one without default values that I wanted to update in some situations, but update_control() with default values is not working for group border. I can't manage to discover how to update the default values using update_control(). The following code doesn't work: 
`$this->update_control('item_border', [
	'default' => [
		'border' => 'solid',
		'color' => '#EEE',
		'width' => [
			'default' => [
				'top' => '1',
				'right' => '1',
				'bottom' => '1',
				'left' => '1',
				'isLinked' => true,
			],
		],
	]
]);`

Any suggestions?

**Comment by perks98** [comment]
@LucasMariniFalbo not sure if you already solved this yourself but I tried to look into myself and the way I got it to update a sub-control to a group is like the following code:
```
$this->update_control( 'item_border_border', [
		'default' => 'dashed',
	]
);
```
So you named that group `item_border`, so from checking the variable stack each sub-control uses that name as a prefix to the final control. So the example above would update the border type (key of `border` making it `item_border_border`) to dashed.
